proto/message_contents/signature.proto

```diff
@@ -27,3 +30,59 @@ message Signature {
         WalletECDSACompact wallet_ecdsa_compact = 2;
     }
 }
+
+// The purpose of the key being linked to an account
+enum AccountLinkedRole {
+    ACCOUNT_LINKED_ROLE_UNSPECIFIED = 0;
+    ACCOUNT_LINKED_ROLE_INBOX_KEY = 1;
+    ACCOUNT_LINKED_ROLE_SEND_KEY = 2;
+}
+
+// A signature used to link a key to an account with a
+// static text format, e.g.
+// 'XMTP : Grant Send Permissions ...'
+// The key_bytes and role must be validated against the
+// signature text
+message AccountLinkedStaticSignature {
+    // V1
+    message V1 {
+        AccountLinkedRole role = 1;
+        bytes text = 2;
+        Signature signature = 3;
+    }
+    oneof version {
+        V1 v1 = 1;
+    }
+}
+
+// A signature used to revoke a key from an account
+// with a static text format, e.g.
+// 'XMTP : Revoke Send Permissions ...'
+// The key_bytes and revocation action must be validated
+// against the signature text
+message AccountRevokedStaticSignature {
+    // V1
+    message V1 {
+        bytes text = 1;
+        Signature signature = 2;
+    }
+    oneof version {
+        V1 v1 = 1;
+    }
+}
+
+// A signature used to link a key to an account, with
+// sign-in with Ethereum format (EIP-4361).
+// The key_bytes and role must be validated
+// against the signature text
+message AccountLinkedSIWESignature {
+    // V1
+    message V1 {
+        AccountLinkedRole role = 1;
```

Comment:
Can you explain how this mechanism works?

If the intention is to use SIW* resource lines to define keys. Isn't the relationship between Signature and Role 1:M?

Is there a need that this is captured when persisted rather than determined during validation? i.e. a Signature is validated for a specific role at runtime? e.g.`signature.validate(inputBytes, ROLE_SEND) -> false`